src/core/lib/deck.js

```diff
@@ -51,13 +64,56 @@ const propTypes = {
   onLayerHover: PropTypes.func,
   useDevicePixels: PropTypes.bool,
 
-  // Debug settings
+  // Viewport props (TODO - should only support these on the react component)
+  longitude: PropTypes.number, // The longitude of the center of the map.
+  latitude: PropTypes.number, // The latitude of the center of the map.
+  zoom: PropTypes.number, // The tile zoom level of the map.
+  bearing: PropTypes.number, // Specify the bearing of the viewport
+  pitch: PropTypes.number, // Specify the pitch of the viewport
+  altitude: PropTypes.number, // Altitude of camera. Default 1.5 "screen heights"
+  position: PropTypes.array, // Camera position for FirstPersonViewport
+
+  // Viewport constraints
+  // TODO - too many props, define constraints object
+  maxZoom: PropTypes.number, // Max zoom level
+  minZoom: PropTypes.number, // Min zoom level
+  maxPitch: PropTypes.number, // Max pitch in degrees
+  minPitch: PropTypes.number, // Min pitch in degrees
+
+  // onViewStateChange: PropTypes.func, // callback, fires when user interacts with the view
+  onViewportChange: PropTypes.func, // callback, fires when user interacts with the view
+
+  // Viewport transition
+  transitionDuration: PropTypes.number, // transition duration for viewport change
+  transitionInterpolator: PropTypes.object, // ViewportTransitionInterpolator for custom transitions
+  transitionInterruption: PropTypes.number, // type of interruption of current transition on update
+  transitionEasing: PropTypes.func, // easing function
+  onTransitionStart: PropTypes.func, // transition status update functions
+  onTransitionInterrupt: PropTypes.func,
+  onTransitionEnd: PropTypes.func,
+
+  // Enables control event handling
+  scrollZoom: PropTypes.bool, // Scroll to zoom
+  dragPan: PropTypes.bool, // Drag to pan
+  dragRotate: PropTypes.bool, // Drag to rotate
+  doubleClickZoom: PropTypes.bool, // Double click to zoom
+  touchZoomRotate: PropTypes.bool, // Pinch to zoom / rotate
+
+  // Accessor that returns a cursor style to show interactive state
+  getCursor: PropTypes.func,
+
+  // Debug props
   debug: PropTypes.bool,
   drawPickingColors: PropTypes.bool
 };
 
-const defaultProps = {
+const defaultProps = Object.assign({}, TransitionManager.defaultProps, {
   id: 'deckgl-overlay',
+
+  // Size
+  width: DEFAULT_WINDOW_SIZE,
```

Comment:
Given width, height are controlled by `canvas`, should we make these private members instead listing them under props?

Reply:
> Is this necessary when props.width and props.height are specified?

> Given width, height are controlled by canvas, should we make these private members instead listing them under props?

Good points. This PR is definitely not the last word on component sizing. I think we may want to support (at least) two modes:
* explicit width&height provided - forces the canvas to that size. At least for backwards compatibility.
* no width&height provided - read width and height from canvas.

Also, ideally the corresponding change should also be done in react-map-gl (otherwise the app will need to track window size anyway, defeating this feature) so we should come up with a good spec that works for both repos.